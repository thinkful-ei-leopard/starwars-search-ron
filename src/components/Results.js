import React, {Component} from 'react';
import ResultCharacter from './ResultCharacter';
import ApiContext from '../ApiContext';
import './Results.css';

export default class Results extends Component {
    // gets data from state
    // iterates over the list to create an individual Result 
    // each Result renders the data.

    static contextType = ApiContext;

    // perhaps for bonus, you could different components to render different data
    // ie ResultCharacter, ResultPlanet, ResultSpaceship - which all probably have different data in them.
    // then conditionally render depending a new value in state that lets us know what type of search it was (ie PATH in Form.js)
    render() {
        console.log(this.context.characters);
        return(
            <ul className="Results__container">
                {this.context.characters.map((result, idx) => {
                    return <ResultCharacter 
                                key={idx} 
                                name={result.name}
                                hairColor={result.hair_color}
                                eyeColor={result.eye_color}
                                birthYear={result.birth_year}
                                gender={result.gender}
                                />
                })}
            </ul>
        );
    }

}