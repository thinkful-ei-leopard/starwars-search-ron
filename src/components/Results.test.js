
import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Results from './Results';

describe('<Results />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Results />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<Results />).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});