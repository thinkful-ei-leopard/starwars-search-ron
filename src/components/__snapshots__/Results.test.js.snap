// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> renders this UI as expected 1`] = `
<ul
  className="Results__container"
/>
`;
